lib/ansible/modules/windows/win_snmp.ps1

```diff
@@ -0,0 +1,126 @@
+#!powershell
+
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$params      = Parse-Args -arguments $args -supports_check_mode $true;
+$check_mode  = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+$managers    = Get-AnsibleParam -obj $params -name "permitted_managers"  -type "list" -default $null
+$communities = Get-AnsibleParam -obj $params -name "community_strings"   -type "list" -default $null
+$action_in   = Get-AnsibleParam -obj $params -name "action"              -type "str"  -default "set" -ValidateSet @("set", "add", "remove")
```

Comment:
Just wondering if these could be expressed as more usual (in ansible modules) 'state: present' /'state: absent'.  This makes the module parameters more a description of the desired state, rather than an operation to be performed.

Reply:
hmm, just noticed comment above asking for the action set/add/remove so feel free to ignore my comment above.